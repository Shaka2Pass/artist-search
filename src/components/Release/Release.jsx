//http://musicbrainz.org/ws/2/recording?release=<RELEASE_ID>&fmt=json
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const Release = ({ title, image, id }) => (
  <section>
    <h2><Link to= {`/recording/${id}`}>{title}</Link></h2>
    <img src={image} alt="Image of album or relsease"></img>
  </section>
);

Release.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Release;


