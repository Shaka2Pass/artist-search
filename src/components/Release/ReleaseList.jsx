import Release from '../Release/Release.jsx';
import withList from '../withList';

const ReleaseList = withList(Release, 'releaseResults');

export default ReleaseList;
