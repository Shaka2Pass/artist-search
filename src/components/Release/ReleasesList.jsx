import Release from '../Release/Release.jsx';
import withList from '../withList';

const ReleasesList = withList(Release, 'releaseResults');

export default ReleasesList;
