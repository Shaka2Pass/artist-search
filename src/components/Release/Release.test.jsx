import React from  'react';
import { shallow } from 'enzyme';
import Release from './Release.jsx';

describe('Release', () => {
  it('Matches a snapshot of the Release component', () => {
    const wrapper = shallow (<Release
      title="Gloria"
      image="http://coverartarchive.org/release/03be9452-549a-4f30-a96f-44f794d50253/front"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});

