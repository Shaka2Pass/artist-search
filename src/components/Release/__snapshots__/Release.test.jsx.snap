// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release Matches a snapshot of the Release component 1`] = `
<section>
  <h2>
    <Link
      to="title/Gloria"
    >
      Gloria
    </Link>
    Gloria
  </h2>
  <img
    alt="Image of album or relsease"
    src="http://coverartarchive.org/release/03be9452-549a-4f30-a96f-44f794d50253/front"
  />
</section>
`;
